x-pack/plugins/spaces/server/spaces_client/spaces_client.ts

```diff
@@ -136,6 +147,14 @@ export class SpacesClient implements ISpacesClient {
     return this.transformSavedObjectToSpace(updatedSavedObject);
   }
 
+  public createSavedObjectFinder(id: string) {
+    return this.repository.createPointInTimeFinder({
+      type: 'visualization', // TODO: find all types
```

Comment:
`repository.createSavedObjectFinder` and `repository.find` require `type` param to be specified and they don't accept a wildcard. How are we expected to retrieve saved objects irregardless of type?

Reply:
You need to use the [`ISavedObjectTypeRegistry`](https://github.com/elastic/kibana/blob/221ee74c6b1573926d5354b3faf8a4e96b58718c/src/core/server/saved_objects/saved_objects_type_registry.ts#L17) to:
1. Get all hidden types and instantiate the SOR with them (otherwise it will ignore them)
2. Get all types (hidden + not hidden) to use in your PIT finder

You can exclude "global" types (`namespaceType: 'agnostic'`) because they exit outside of Spaces.
Here's a diff of how you could do this:

```diff
diff --git a/x-pack/plugins/spaces/server/spaces_client/spaces_client.ts b/x-pack/plugins/spaces/server/spaces_client/spaces_client.ts
index b93ba3c24e5..561040f8c7a 100644
--- a/x-pack/plugins/spaces/server/spaces_client/spaces_client.ts
+++ b/x-pack/plugins/spaces/server/spaces_client/spaces_client.ts
@@ -89,7 +89,8 @@ export class SpacesClient implements ISpacesClient {
   constructor(
     private readonly debugLogger: (message: string) => void,
     private readonly config: ConfigType,
-    private readonly repository: ISavedObjectsRepository
+    private readonly repository: ISavedObjectsRepository,
+    private readonly nonGlobalTypeNames: string[]
   ) {}
 
   public async getAll(options: GetAllSpacesOptions = {}): Promise<GetSpaceResult[]> {
@@ -149,7 +150,7 @@ export class SpacesClient implements ISpacesClient {
 
   public createSavedObjectFinder(id: string) {
     return this.repository.createPointInTimeFinder({
-      type: 'visualization', // TODO: find all types
+      type: this.nonGlobalTypeNames,
       namespaces: [id],
       perPage: 10_000,
     });
diff --git a/x-pack/plugins/spaces/server/spaces_client/spaces_client_service.ts b/x-pack/plugins/spaces/server/spaces_client/spaces_client_service.ts
index 6580a2d57f0..e2ba0d92111 100644
--- a/x-pack/plugins/spaces/server/spaces_client/spaces_client_service.ts
+++ b/x-pack/plugins/spaces/server/spaces_client/spaces_client_service.ts
@@ -96,9 +96,16 @@ export class SpacesClientService {
   }
 
   public start(coreStart: CoreStart): SpacesClientServiceStart {
+    const nonGlobalTypes = coreStart.savedObjects
+      .getTypeRegistry()
+      .getAllTypes()
+      .filter((x) => x.namespaceType !== 'agnostic');
+
     if (!this.repositoryFactory) {
+      const hiddenTypeNames = nonGlobalTypes.filter((x) => x.hidden).map((x) => x.name);
+      const includedHiddenTypes = [...hiddenTypeNames, 'space'];
       this.repositoryFactory = (request, savedObjectsStart) =>
-        savedObjectsStart.createScopedRepository(request, ['space']);
+        savedObjectsStart.createScopedRepository(request, includedHiddenTypes);
     }
     return {
       createSpacesClient: (request: KibanaRequest) => {
@@ -106,10 +113,12 @@ export class SpacesClientService {
           throw new Error('Initialization error: spaces config is not available');
         }
 
+        const nonGlobalTypeNames = nonGlobalTypes.map((x) => x.name);
         const baseClient = new SpacesClient(
           this.debugLogger,
           this.config,
-          this.repositoryFactory!(request, coreStart.savedObjects)
+          this.repositoryFactory!(request, coreStart.savedObjects),
+          nonGlobalTypeNames
         );
         if (this.clientWrapper) {
           return this.clientWrapper(request, baseClient);
```

Reply:
Is there a way for plugins to dynamically register new saved object types which this service could miss if it only fetches them during start phase?

Reply:
> Is there a way for plugins to dynamically register new saved object types which this service could miss if it only fetches them during start phase?

Good question -- no, object types can only be registered during plugin `setup`, so we don't have to worry about this situation!